### ISSUE TITLE
Handling Multiple Props

### ISSUE THREAD

**Issue by ghost** [scf4/styled-map#24]
I have a Button component which I have mapped to a buttons object.  In that object, I have a color section, with various options:  `primary`, `secondary`, etc.  Like this:
```
color: {
    default: {
      backgroundColor: primary,
      borderColor: primary,
      color: #fff
      }
    },
    primary: {
      backgroundColor: primary,
      borderColor: primary,
      color: #fff
    },
    secondary: {
      backgroundColor: secondary,
      borderColor: secondary,
      color: #fff
    },
```

In my button component, I include this like this:   

```
const Button = styled('button')`
...
${theme('buttons.color')}
...
`;
```

I can now create a button like this:  `<Button primary>Click Me</Button>`.

And it works just like I want.

But now I want to add a second prop: `outline`.  Like this:  `<Button primary outline>Click Me</Button>`.

This should change the `background-color` to `transparent` and the `color` property to whatever prop I passed in from the `color` section of the buttons object.  

So, for instance, `<Button primary outline>Click Me</Button>` would produce the following css:

```
   background-color: transparent;
   border-color: primary;
   color: primary;
```

Whereas `<Button secondary outline>Click Me</Button>` would produce the following css:

```
   background-color: transparent;
   border-color: secondary;
   color: secondary;
```

The problem is that I can't figure out how to set the `color` property to `primary`, `secondary`. 

In short, I don't know to handle multiple props where those props need to work in conjunction with each other.

Any ideas how to do this?

Thanks.

**Comment by scf4** [comment]
See README.md
